expose.go

```diff
@@ -0,0 +1,117 @@
+package chromedp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/chromedp/cdproto/page"
+	"github.com/chromedp/cdproto/runtime"
+)
+
+const (
+	deliverError     = "deliverError"
+	deliverResult    = "deliverResult"
+	addTargetBinding = "addTargetBinding"
+)
+
+// BindingCalledPayload ...
+type BindingCalledPayload struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
+	Seq  int64  `json:"seq"`
+	Args string `json:"args"`
+}
+
+// BindingFunc expose function type
+type BindingFunc func(args string) (string, error)
+
+// ExposeFunc The method adds a function called name on the browser page's window object.
+// When called, the function executes BindingFunc in go env
+// and returns a Promise which resolves to the return value of BindingFunc.
+// Note. compared with puppeteer's exposeFunction.
+// the BindingFunc takes exactly one argument, this argument should be string
+func ExposeFunc(ctx context.Context, fnName string, fn BindingFunc) error {
+	c := FromContext(ctx)
+	if c == nil {
+		return ErrInvalidContext
+	}
+	if c.Target == nil {
+		if err := c.newTarget(ctx); err != nil {
+			return err
+		}
+	}
+
+	c.Target.bindingFuncListenOnce.Do(func() {
```

Comment:
If `ExposeFunc` is an Action, is it ok to remove `bindingFuncListenOnce.Do`? Please confirm whether `addTargetBinding` can be defined with `page.AddScriptToEvaluateOnNewDocument` multiple times.

Reply:
By modifying the code, it can be executed multiple times, but the effect is equivalent to function coverage